dev/com.ibm.ws.config/src/com/ibm/ws/config/admin/internal/ConfigurationStorageHelper.java

```diff
@@ -0,0 +1,673 @@
+        return location;
+
+    }
+
+    private static String readPossiblyNullString(DataInputStream dis) throws IOException {
+        if (dis.readBoolean()) {
+            return dis.readUTF();
+        }
+        return null;
+    }
+
+    public static void store(File configFile, Collection<? extends ExtendedConfiguration> configs) throws IOException {
+        try (DataOutputStream dos = new DataOutputStream(new BufferedOutputStream(new FileOutputStream(configFile, false)))) {
+            dos.writeByte(VERSION);
+            dos.writeInt(configs.size());
+            for (ExtendedConfiguration config : configs) {
+                Dictionary<String, ?> properties = config.getReadOnlyProperties();
+                Set<ConfigID> references = config.getReferences();
+                Set<String> uniqueVars = config.getUniqueVariables();
+                String bundleLocation = config.getBundleLocation();
+                store(dos, properties, bundleLocation, references, uniqueVars);
+            }
+        }
+    }
+
+    public static void store(DataOutputStream dos, Dictionary<String, ?> properties, String bundleLocation,
+                             Set<ConfigID> references, Set<String> uniqueVars) throws IOException {
+
+        writePossiblyNullString(dos, bundleLocation);
+        if (uniqueVars != null) {
+            dos.writeInt(uniqueVars.size());
+            for (String var : uniqueVars) {
+                dos.writeUTF(var);
+            }
+        } else {
+            dos.writeInt(0);
+        }
+        if (references != null) {
+            dos.writeInt(references.size());
+            int count = 0;
+            Map<ConfigID, Integer> writtenConfigIds = new HashMap<>();
+            for (ConfigID id : references) {
+                count = writeConfigID(dos, id, count, 0, writtenConfigIds);
+            }
+        } else {
+            dos.writeInt(0);
+        }
+        writeMap(dos, toMapOrDictionary(properties));
+    }
+
+    public static void readMap(DataInputStream dis, MapIterable props) throws IOException {
+        int size = dis.readInt();
+        for (int i = 0; i < size; i++) {
+            String key = dis.readUTF();
+            Object value = readValue(dis);
+            props.put(key, value);
+        }
+    }
+
+    private static Object readValue(DataInputStream dis) throws IOException {
+        byte type = dis.readByte();
+        Object value;
+
+        if (type == NULL) {
+            value = null;
+        } else if (type == COLLECTION) {
+            value = readCollection(dis);
+        } else if (type == ARRAY) {
+            value = readArray(dis);
+        } else if (type == MAP) {
+            Map<String, Object> map = new HashMap<>();
+            value = map;
+            readMap(dis, toMapOrDictionary(map));
+        } else {
+            value = readPrimative(dis, type);
+        }
+
+        return value;
+    }
+
+    // TODO how to handle null values?
```

Comment:
does this TODO still apply?

Reply:
Opps, I still need to address this.  But I'm not sure we allow for arrays.  It doesn't seem like it.

Reply:
Fixed this with the last commit to handle null values in arrays and mixed type collections